packages/store-sync/src/createStoreSync.ts

```diff
@@ -215,18 +228,30 @@ export async function createStoreSync({
           ? bigIntMax(range.startBlock, lastBlockNumberProcessed + 1n)
           : range.startBlock,
         toBlock: range.endBlock,
-        storageAdapter,
         logFilter,
       });
```

Comment:
one of the reasons I originally coupled the fetch+store operation is so that the fetching wouldn't get ahead of the storing of logs, creating a potential memory explosion

do you know if that's a concern here in the way rxjs processes the stream?

Reply:
https://playcode.io/2019292 -> if storing the logs takes longer than fetching the logs, it would indeed buffer the logs in rxjs and potentially increase memory more than if we only fetch when the previous logs were applied. Wonder if this is actually significant here though - when loading the snapshot from the indexer we already keep the entire snapshot in memory while it's being fed into the storage adapter and the RPC sync wouldn't produce a larger in-memory backlog than that (especially not during live sync, if anything only during the initial sync from RPC, but even in that case it shouldn't be much more than the snapshot).

Reply:
iirc this was mostly an issue in indexers where you are fetching the entire chain state (many worlds from block 0) and fetching was much faster than storing into the DB, so the process would repeatedly get OOM-killed

Reply:
reverted to `fetchAndStoreLogs` and storing logs separately in the indexer logs stream